wp/lib/bap_wp/src/output.ml

```diff
@@ -180,20 +180,28 @@ let print_result ?fmt:(fmt = Format.err_formatter) (solver : Solver.solver)
     Format.fprintf fmt "\nModel:\n%s\n%!" (format_model model env1 env2);
     let print_refuted_goals = List.mem show "refuted-goals" ~equal:String.equal in
     let print_path = List.mem show "paths" ~equal:String.equal in
-    (* If 'paths' is specified, we assume we are also printing the refuted goals. *)
-    if print_refuted_goals || print_path then begin
+    (* Only compute the refuted goals when we print them or write CFGs *)
+    if print_refuted_goals || print_path || Option.is_some dump_cfgs then
       let var_map1 = Env.get_var_map env1 in
       let var_map2 = Env.get_var_map env2 in
       let mem1, _ = Env.get_var env1 (Env.get_mem env1) in
       let mem2, _ = Env.get_var env2 (Env.get_mem env2) in
       let refuted_goals =
         Constr.get_refuted_goals goals solver ctx ~filter_out:[mem1; mem2] in
-      Format.fprintf fmt "%s%!" "\nRefuted goals:\n";
-      Seq.iter refuted_goals ~f:(fun goal ->
-          Format.fprintf fmt "%s\n%!"
-            (Constr.format_refuted_goal goal model ~orig:(var_map1, sub1)
-               ~modif:(var_map2, sub2) ~print_path))
-    end
+      begin
+        (* If 'paths' is specified, we assume we are also printing the refuted goals. *)
+        if print_refuted_goals || print_path then
+          begin 
+            Format.fprintf fmt "%s%!" "\nRefuted goals:\n";
+            Seq.iter refuted_goals ~f:(fun goal ->
+                Format.fprintf fmt "%s\n%!"
+                  (Constr.format_refuted_goal goal model ~orig:(var_map1, sub1)
+                     ~modif:(var_map2, sub2) ~print_path))
+          end;
+        Option.iter dump_cfgs ~f:(fun fname -> 
```

Comment:
After some though, I think the `dump_cfgs` option is unneeded. We can instead rely on the behavior of `--show=paths`, where `fname` is then implicitly the name of the subroutine that we're doing analysis on.

Reply:
There's a mapping from original to modified subroutine names. If we ever see a case where the original and modified names are different, we can use the mapping to produce the correct name for the file with the DOT representation of the modified CFG.